### PR TITLE
test: more tests and cleanup

### PR DESCRIPTION
**Pull request by fmorency**
This PR adds
- Additional keeper unit tests
- Check error content in tests
- Code cleanup (value vs pointer receiver mix, comments, lint)
- bump go to 1.22.4 (security fixes)
- bump alpine to 3.20 otherwise heighliner fails

### CODE REVIEW

**Comment by fmorency** [comment]
I'll rerun the failed job tomorrow as we reached Docker request limits.

`toomanyrequests: You have reached your pull rate limit.`

**Comment by fmorency** [comment]
Disregard the codecov/patch complain.

File: Dockerfile [hunk 1]
```diff
@@ -28,7 +28,7 @@ RUN LEDGER_ENABLED=false BUILD_TAGS=muslc LINK_STATICALLY=true make $BUILD_CMD \
   && (file /code/build/manifestd | grep "statically linked")
 
 # --------------------------------------------------------
-FROM alpine:3.16
+FROM alpine:3.20
```
Comment (Reecepbcups) [inline Dockerfile:31]: ahh yea we were on a lower version when cosmwasm was being used early days

File: x/manifest/depinject.go [hunk 2]
```diff
@@ -22,10 +22,10 @@ import (
 var _ appmodule.AppModule = AppModule{}
 
 // IsOnePerModuleType implements the depinject.OnePerModuleType interface.
-func (am AppModule) IsOnePerModuleType() {}
+func (a AppModule) IsOnePerModuleType() {}
```
Comment (Reecepbcups) [inline x/manifest/depinject.go:25]: convention is am but this is minor (SDK does not follow go best practice)

File: x/manifest/types/codec.go [hunk 3]
```diff
@@ -3,19 +3,10 @@ package types
...
 	"github.com/cosmos/cosmos-sdk/codec"
 	"github.com/cosmos/cosmos-sdk/codec/types"
-	cryptocodec "github.com/cosmos/cosmos-sdk/crypto/codec"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/types/msgservice"
 )
 
-var amino = codec.NewLegacyAmino()
-
-func init() {
```
Comment (Reecepbcups) [inline x/manifest/types/codec.go:13]: I remember removing this in the past for a network & it causing issues. They did come up from SDK v47 -> v50 though, so it's possible it was specific to that vs being on SDK v50 from the start like manifest. Just want to call it out now

**Review by Reecepbcups — APPROVED** [review]

File: x/manifest/types/codec.go (hunk 3 above)
Comment (fmorency) [reply x/manifest/types/codec.go:13]: Do you remember the issue?

File: x/manifest/depinject.go (hunk 2 above)
Comment (fmorency) [reply x/manifest/depinject.go]: [32d5026](https://github.com/liftedinit/manifest-ledger/pull/69/commits/32d5026fd1cd96fe569b13b7cfa187ece0dc57fb)

**Pull request merged**